awx/api/views/__init__.py

```diff
@@ -2267,65 +2234,44 @@ def modernize_launch_payload(self, data, obj):
-                ('vault_credential', lambda cred: cred.credential_type.kind != 'vault', int, 'pk value'),
-                ('extra_credentials', lambda cred: cred.credential_type.kind not in ('cloud', 'net'), Iterable, 'a list')
-            ):
-                if key in modern_data:
-                    # if a specific deprecated key is specified, remove all
-                    # credentials of _that_ type from the list of current
-                    # credentials
-                    existing_credentials = filter(conditional, existing_credentials)
-                    prompted_value = modern_data.pop(key)
-
-                    # validate type, since these are not covered by a serializer
-                    if not isinstance(prompted_value, _type):
-                        msg = _(
-                            "Incorrect type. Expected {}, received {}."
-                        ).format(type_repr, prompted_value.__class__.__name__)
-                        raise ParseError({key: [msg], 'credentials': [msg]})
-
-                    # add the deprecated credential specified in the request
-                    if not isinstance(prompted_value, Iterable) or isinstance(prompted_value, str):
-                        prompted_value = [prompted_value]
-
-                    # If user gave extra_credentials, special case to use exactly
-                    # the given list without merging with JT credentials
-                    if key == 'extra_credentials' and prompted_value:
-                        obj._deprecated_credential_launch = True  # signal to not merge credentials
-                    new_credentials.extend(prompted_value)
+            if 'extra_credentials' in modern_data:
+                existing_credentials = [
+                    cred for cred in existing_credentials
+                    if cred.credential_type.kind not in ('cloud', 'net')
+                ]
+                prompted_value = modern_data.pop('extra_credentials')
+
+                # validate type, since these are not covered by a serializer
+                if not isinstance(prompted_value, Iterable):
+                    msg = _(
+                        "Incorrect type. Expected a list received {}."
+                    ).format(prompted_value.__class__.__name__)
+                    raise ParseError({'extra_credentials': [msg], 'credentials': [msg]})
+
+                # add the deprecated credential specified in the request
+                if not isinstance(prompted_value, Iterable) or isinstance(prompted_value, str):
+                    prompted_value = [prompted_value]
+
+                # If user gave extra_credentials, special case to use exactly
+                # the given list without merging with JT credentials
+                if prompted_value:
+                    obj._deprecated_credential_launch = True  # signal to not merge credentials
+                new_credentials.extend(prompted_value)
```

Comment:
this removal of the loop looks good on first read. It'll need some more attention when we get closer to merge.